### ISSUE TITLE
SARIMAX array broadcast ValueError

### ISSUE THREAD

**Issue by tgsmith61591** [statsmodels/statsmodels#3799]
I'm getting a `ValueError` in the `SARIMAX` class, and I'm not sure whether it's a bug or my data... if it's my data, there should be a more informative error message.

Here is my version:

```python
>>> sm.version.full_version
'0.8.0'
```

Here's the setup:

```python
from statsmodels import api as sm
import numpy as np

# define the data
wineind = np.array([15136, 16733, 20016, 17708, 18019, 19227, 22893, 23739, 21133,
                    22591, 26786, 29740, 15028, 17977, 20008, 21354, 19498, 22125,
                    25817, 28779, 20960, 22254, 27392, 29945, 16933, 17892, 20533,
                    23569, 22417, 22084, 26580, 27454, 24081, 23451, 28991, 31386,
                    16896, 20045, 23471, 21747, 25621, 23859, 25500, 30998, 24475,
                    23145, 29701, 34365, 17556, 22077, 25702, 22214, 26886, 23191,
                    27831, 35406, 23195, 25110, 30009, 36242, 18450, 21845, 26488,
                    22394, 28057, 25451, 24872, 33424, 24052, 28449, 33533, 37351,
                    19969, 21701, 26249, 24493, 24603, 26485, 30723, 34569, 26689,
                    26157, 32064, 38870, 21337, 19419, 23166, 28286, 24570, 24001,
                    33151, 24878, 26804, 28967, 33311, 40226, 20504, 23060, 23562,
                    27562, 23940, 24584, 34303, 25517, 23494, 29095, 32903, 34379,
                    16991, 21109, 23740, 25552, 21752, 20294, 29009, 25500, 24166,
                    26960, 31222, 38641, 14672, 17543, 25453, 32683, 22449, 22316,
                    27595, 25451, 25421, 25288, 32568, 35110, 16052, 22146, 21198,
                    19543, 22084, 23816, 29961, 26773, 26635, 26972, 30207, 38687,
                    16974, 21697, 24179, 23757, 25013, 24019, 30345, 24488, 25156,
                    25650, 30923, 37240, 17466, 19463, 24352, 26805, 25236, 24735,
                    29356, 31234, 22724, 28496, 32857, 37198, 13652, 22784, 23565,
                    26323, 23779, 27549, 29660, 23356])

# the order of differencing
d = D = 1
m = 12
```

And here's where the error occurs:

```python
>>> arima = sm.tsa.statespace.SARIMAX(endog=wineind, exog=None, order=(0,d,0), seasonal_order=(0,D,0,m))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "//anaconda/lib/python2.7/site-packages/statsmodels/tsa/statespace/sarimax.py", line 524, in __init__
    self.ssm.transition = self.initial_transition
  File "//anaconda/lib/python2.7/site-packages/statsmodels/tsa/statespace/sarimax.py", line 795, in initial_transition
    transition[start:end, start:end] = companion_matrix(self._k_order)
ValueError: could not broadcast input array from shape (0,0) into shape (13,13)
```

**Comment by bashtage** [comment]
Appears to work with master, which suggest this was fixed after 0.8 release.

```
arima.fit().summary()
Out[5]: 
<class 'statsmodels.iolib.summary.Summary'>
"""
                                 Statespace Model Results                                 
==========================================================================================
Dep. Variable:                                  y   No. Observations:                  176
Model:             SARIMAX(0, 1, 0)x(0, 1, 0, 12)   Log Likelihood               -1564.704
Date:                            Thu, 29 Jun 2017   AIC                           3131.408
Time:                                    20:42:29   BIC                           3134.502
Sample:                                         0   HQIC                          3132.664
                                            - 176                                         
Covariance Type:                              opg                                         
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
sigma2      1.268e+07   1.09e+06     11.674      0.000    1.05e+07    1.48e+07
===================================================================================
Ljung-Box (Q):                      129.94   Jarque-Bera (JB):                11.59
Prob(Q):                              0.00   Prob(JB):                         0.00
Heteroskedasticity (H):               3.17   Skew:                            -0.03
Prob(H) (two-sided):                  0.00   Kurtosis:                         4.31
===================================================================================

Warnings:
[1] Covariance matrix calculated using the outer product of gradients (complex-step).
"""
```

**Comment by ChadFulton** [comment]
Yes, this was fixed in #3383. Thanks for the report, though!
